src/ddsrt/src/library/windows/library.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright(c) 2006 to 2018 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#include <stdio.h>
+#include <assert.h>
+#include <string.h>
+#include "dds/ddsrt/heap.h"
+#include "dds/ddsrt/types.h"
+#include "dds/ddsrt/string.h"
+#include "dds/ddsrt/library.h"
+
+
+dds_retcode_t
+ddsrt_dlopen(
+    const char *name,
+    bool translate,
+    ddsrt_lib *handle)
+{
+    dds_retcode_t retcode = DDS_RETCODE_OK;
+
+
+    if ( name == NULL || handle == NULL ){
+        retcode = DDS_RETCODE_BAD_PARAMETER;
```

Comment:
early return is not a good practice, I try single return as much as I can. If you have a previous decision, I can change it.
I remember this practice in JOINT STRIKE FIGHTER AIR VEHICLE C++ CODING STANDARDS 

> AV Rule 113 (MISRA Rule 82, Revised)
Functions will have a single exit point.
Rationale: Numerous exit points tend to produce functions that are both difficult to
understand and analyze.
Exception: A single exit is not required if such a structure would obscure or otherwise
significantly complicate (such as the introduction of additional variables) a function’s control
logic. Note that the usual resource clean-up must be managed at all exit points.

Reply:
@kurtuluso : please read what you quoted:

> A single exit is not required if such a structure would obscure or otherwise
significantly complicate (such as the introduction of additional variables) a function’s control
logic. Note that the usual resource clean-up must be managed at all exit points.

An early return here reduces the amount of state on needs to track mentally (no null pointers — so no need to account for null pointers anywhere in the function anymore). No state has been changed yet, so an early return is the better option.

Reply:
It is ~20 lines of function in total. Using a single return here does not make it complicated or introduce new additional variables. Of course, early return always reduces the amount of state on needs to track mentally but I think it does not fit the exception.
Anyway, I will not continue the discussion anymore, I'm just adding a return.